code/replication/object/Procedure.py

```diff
@@ -0,0 +1,76 @@
+from Similarity import *
+from BSBManager import *
+from Article import Article
+import time
+import glob
+import sys
+from multiprocessing import cpu_count
+import csv
+from ETUtils import *
+
+def procedure(startlocation = 'data', endlocation='export_dataframe.csv', simtest=CosineSimilarity(), worker_count=-1):
+	'''
+	Performs the procedure for the specified amount of articles. Uses 
+	all nml files from startlocation, and exports a csv file at endlocation.
+	'''
+
+	if worker_count < 0:
+		worker_count += cpu_count() + 1
+
+	worker_count = worker_count * 3
```

Comment:
These types of decisions should be documented in the code. I remember us talking about it, but good to explain to others...